src/PrestaShopBundle/Controller/Admin/CommonController.php

```diff
@@ -56,20 +61,52 @@ class CommonController extends FrameworkBundleAdminController
      */
     public function notificationsAction()
     {
-        // TODO: Use CQRS
-        return new JsonResponse((new \Notification())->getLastElements());
+        $employeeId = Context::getContext()->employee->id;
+        /** @var NotificationsResults $elements */
+        $elements = $this->getQueryBus()->handle(new GetNotificationLastElements($employeeId));
+
+        $response = [];
+
+        foreach ($elements->getNotificationsResults() as $element) {
```

Comment:
this should not be done in the controller at all imho how about create an object with this responsibility ?